### PR TITLE
Add Financial Phrasebank dataset

### PR DESCRIPTION
**Pull request by dipam7**
## Checklist for adding MMTEB dataset

Reason for dataset addition:

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] I have considered the size of the dataset and reduced it if it is too big (2048 examples is typically large enough for most tasks)
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

File: mteb/tasks/Classification/eng/FinancialPhrasebankClassification.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            @article{Malo2014GoodDO,
+            title={Good debt or bad debt: Detecting semantic orientations in economic texts},
+            author={P. Malo and A. Sinha and P. Korhonen and J. Wallenius and P. Takala},
+            journal={Journal of the Association for Information Science and Technology},
+            year={2014},
+            volume={65}
+            }
+        """,
+        n_samples={"train": 4840},
+        avg_character_length=None,  # Optional: You can calculate this based on the dataset if needed
```
Comment (isaac-chung) [inline mteb/tasks/Classification/eng/FinancialPhrasebankClassification.py:42]: https://huggingface.co/datasets/financial_phrasebank/viewer/sentences_allagree's train split has 2.26k rows. 
Here is a quick way to calculate avg char len: 
```
import numpy as np
np.mean(list(map(len, dataset["train"]['text'])))
```

File: mteb/tasks/Classification/eng/FinancialPhrasebankClassification.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    )
+
+    @property
+    def metadata_dict(self) -> dict[str, str]:
+        metadata_dict = dict(self.metadata)
+        metadata_dict["n_experiments"] = 10
+        metadata_dict["samples_per_label"] = 128
+        return metadata_dict
```
Comment (isaac-chung) [inline mteb/tasks/Classification/eng/FinancialPhrasebankClassification.py:50]: Can be omitted.

File: mteb/tasks/Classification/eng/FinancialPhrasebankClassification.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    def dataset_transform(self):
+        self.dataset = self.dataset.map(
+            lambda examples: {"text": examples["sentence"]}, remove_columns=["sentence"]
+        )
```
Comment (isaac-chung) [inline mteb/tasks/Classification/eng/FinancialPhrasebankClassification.py:55]:
```suggestion
        self.dataset = self.dataset.rename_column("sentence", "text")
        )
```

File: mteb/tasks/Classification/eng/FinancialPhrasebankClassification.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        eval_langs=["eng-Latn"],
+        main_score="accuracy",
+        date=None,
+        form=None,
+        domains=["News"],
+        task_subtypes=None,
+        license="cc-by-nc-sa-3.0",
+        socioeconomic_status=None,
+        annotations_creators="expert-annotated",
+        dialect=None,
```
Comment (isaac-chung) [inline mteb/tasks/Classification/eng/FinancialPhrasebankClassification.py:30]:
```suggestion
        dialect=[],
```

File: mteb/tasks/Classification/eng/FinancialPhrasebankClassification.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        text_creation=None,
```
Comment (isaac-chung) [inline mteb/tasks/Classification/eng/FinancialPhrasebankClassification.py:31]:
```suggestion
        text_creation="found",
```

File: mteb/tasks/Classification/eng/FinancialPhrasebankClassification.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        },
+        type="Classification",
+        category="s2s",
+        eval_splits=["train"],
+        eval_langs=["eng-Latn"],
+        main_score="accuracy",
+        date=None,
+        form=None,
+        domains=["News"],
+        task_subtypes=None,
```
Comment (isaac-chung) [inline mteb/tasks/Classification/eng/FinancialPhrasebankClassification.py:26]:
```suggestion
        task_subtypes=["Sentiment/Hate speech"],
```

File: mteb/tasks/Classification/eng/FinancialPhrasebankClassification.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+            "revision": "1484d06fe7af23030c7c977b12556108d1f67039",
+            "name": "sentences_allagree",
+        },
+        type="Classification",
+        category="s2s",
+        eval_splits=["train"],
+        eval_langs=["eng-Latn"],
+        main_score="accuracy",
+        date=None,
+        form=None,
```
Comment (isaac-chung) [inline mteb/tasks/Classification/eng/FinancialPhrasebankClassification.py:24]:
```suggestion
        form=["writen"],
```

File: mteb/tasks/Classification/eng/FinancialPhrasebankClassification.py [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+        domains=["News"],
+        task_subtypes=None,
+        license="cc-by-nc-sa-3.0",
+        socioeconomic_status=None,
```
Comment (isaac-chung) [inline mteb/tasks/Classification/eng/FinancialPhrasebankClassification.py:28]: My rough guess for financial news.
```suggestion
        socioeconomic_status="medium",
```

File: mteb/tasks/Classification/eng/FinancialPhrasebankClassification.py [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+            "path": "financial_phrasebank",
+            "revision": "1484d06fe7af23030c7c977b12556108d1f67039",
+            "name": "sentences_allagree",
+        },
+        type="Classification",
+        category="s2s",
+        eval_splits=["train"],
+        eval_langs=["eng-Latn"],
+        main_score="accuracy",
+        date=None,
```
Comment (isaac-chung) [inline mteb/tasks/Classification/eng/FinancialPhrasebankClassification.py:23]: Can just provide a rough estimate here

**Review by isaac-chung — COMMENTED** [review]
Thanks for adding this! just a few small comments.

File: mteb/tasks/Classification/eng/FinancialPhrasebankClassification.py (hunk 9 above)
Comment (dipam7) [reply mteb/tasks/Classification/eng/FinancialPhrasebankClassification.py:23]: I didn't see any dates on huggingface or in the paper. I have no way of estimating it. Also I see a lot of [other datasets](https://github.com/embeddings-benchmark/mteb/blob/edef77bbcce8e0a7a362627b13596391d1a4c3d9/mteb/tasks/Classification/eng/Banking77Classification.py#L22) have set it to None.

**Comment by dipam7** [comment]
Not sure why the test is failing, make test passes for me locally

**Review by isaac-chung — APPROVED** [review]
Looks good. Thanks for iterating!

**Pull request merged**